### PR TITLE
Deprecate the use of spring.cloud.task.batch.listener.enable

### LINKED ISSUE #379: spring.cloud.task.batch.listener.enable is not standard
it should  be `spring.cloud.task.batch.listener.enabled`

### PR DESCRIPTION
**Pull request by cppwfs**
replace it with spring.cloud.task.batch.listener.enabled

resolves #379

### CODE REVIEW

File: spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/configuration/TaskBatchAutoConfiguration.java [hunk 1]
```diff
@@ -32,11 +32,14 @@
 /**
  * Provides auto configuration for the {@link TaskBatchExecutionListener}.
  *
+ * The spring.cloud.task.batch.listener.enable is being deprecated,
```
Comment (mminella) [inline spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/configuration/TaskBatchAutoConfiguration.java]: `is deprecated`...not `is being deprecated` (makes it sound like we're going to get to it later).

File: spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java [hunk 2]
```diff
@@ -86,6 +86,36 @@ public void testAutobuiltDataSource() {
...
+	@Test(expected = AssertionError.class)
+	public void testNoAutoConfigurationBothDisabled() {
+		this.applicationContext = SpringApplication.run(new Class[] {JobConfiguration.class,
+				PropertyPlaceholderAutoConfiguration.class,
+				EmbeddedDataSourceConfiguration.class,
+				BatchAutoConfiguration.class,
+				TaskBatchAutoConfiguration.class}, new String[] {"--spring.cloud.task.batch.listener.enable=false --spring.cloud.task.batch.listener.enabled=false"});
+		validateContext();
+	}
+
```
Comment (mminella) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: Can we get a test where the user explicitly sets the property value to true?

Comment (cppwfs) [reply spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: Forgot to add comment here.  The PR was updated with that test.

**Comment by mminella** [comment]
LGTM.  Rebase, squashed, and merged as b6f6d68385fbec804248a95e5c48948b03153246

**Pull request closed without merging**